### PR TITLE
add hdm cover activity values

### PR DESCRIPTION
**Pull request by sander76**
hdm based blinds (basically all "ease" blinds https://homematic-ip.com/en/product/ease-powered-homematic-ip) seem to report different `ACTIVITY_STATE` values compared to other HMIP blinds. Instead of using "OPENING" and "CLOSING" it uses 1 and 2 integers as displayed in my screenshots below.

opening:
opening

closing:
closing

See my suggested changes. Happy to change if needed.
Also, is there a way to test this locally ? (I have ease blinds available.)

Sander.

### CODE REVIEW

File: hahomematic/platforms/custom/cover.py [hunk 1]
```diff
@@ -145,14 +152,20 @@ def is_closed(self) -> bool | None:
     def is_opening(self) -> bool | None:
         """Return if the cover is opening."""
         if self._e_direction.value is not None:
-            return str(self._e_direction.value) == CoverActivity.OPENING
+            return (
+                str(self._e_direction.value) == CoverActivity.OPENING
+                or self._e_direction.value == HdmCoverActivity.OPENING
```
Comment (sander76) [inline hahomematic/platforms/custom/cover.py:164]: The first (original) evaluation does a cast to `str`. Not sure whether it is wise to cast the new evaluation to an `int` as that might raise an exception?

Comment (SukramJ) [reply hahomematic/platforms/custom/cover.py:164]: mypy loves the cast to int

**Comment by SukramJ** [comment]
Thanks for contributing.
I made some changes (sort enums, extra type for hdm, add test)

Are you able to test it locally, or should i just merge it?

**Comment by SukramJ** [comment]
Hi,

i checked the definition of the HDM1 and ACTIVITY_STATE (used for direction is used like this):

```
      "ACTIVITY_STATE": {
        "MIN": "UNKNOWN",
        "OPERATIONS": 5,
        "MAX": "STABLE",
        "FLAGS": 1,
        "ID": "ACTIVITY_STATE",
        "TYPE": "ENUM",
        "DEFAULT": "UNKNOWN",
        "VALUE_LIST": [
          "UNKNOWN",
          "UP",
          "DOWN",
          "STABLE"
        ]
      }
```

I wonder if this PR is needed.
Has the activity not been displayed correctly yet?

Otherwise, thanks to the stored value_list, the `ACTIVITY_STATE` should also function correctly without the need for an additional enum for the HDM.

**Comment by sander76** [comment]
I have these devices at home so I can check.

Am I assuming correctly the Avtivity_State should be displayed as the state attribute in hass when my shade is either moving up or down? Currently the only states I get are "open" or "close".

**Comment by SukramJ** [comment]
I tested a bit more and the problem is not the misinterpreted `activity_state` .
The problem is a wrong mapping in definitions.py where activity_state is not correctly mapped to direction.

I think this one can be closed.

**Comment by SukramJ** [comment]
See #1430

**Comment by SukramJ** [comment]
Please come back, if next version still has the issue.

**Comment by sander76** [comment]
> Hi,
> 
> i checked the definition of the HDM1 and ACTIVITY_STATE (used for direction is used like this):
> 
> ```
>       "ACTIVITY_STATE": {
>         "MIN": "UNKNOWN",
>         "OPERATIONS": 5,
>         "MAX": "STABLE",
>         "FLAGS": 1,
>         "ID": "ACTIVITY_STATE",
>         "TYPE": "ENUM",
>         "DEFAULT": "UNKNOWN",
>         "VALUE_LIST": [
>           "UNKNOWN",
>           "UP",
>           "DOWN",
>           "STABLE"
>         ]
>       }
> ```
> 
> I wonder if this PR is needed.
> Has the activity not been displayed correctly yet?
> 
> Otherwise, thanks to the stored value_list, the `ACTIVITY_STATE` should also function correctly without the need for an additional enum for the HDM.
> 
> 

So where do you get these values from?

**Comment by SukramJ** [comment]
You can [export the definition](https://github.com/danielperna84/custom_homematic#homematicip_localexport_device_definition) with HA.

We collect these definitions in [pydevccu](https://github.com/danielperna84/pydevccu) to have a virtual CCU for testing.

**Comment by sander76** [comment]
> Please come back, if next version still has the issue.

I will. Thanks!

**Comment by SukramJ** [comment]
Can be tested with 1.56.0b0 .

Beta use can be activated in HACS (Integration/re-download/activate beta)

**Pull request closed without merging**